File: Bazy-Danych/BazyDanych2/Lab4/zadanie domowe/zadanie3.py
from pymongo import MongoClient
from bson.code import Code
db = MongoClient().jeopardy

db.question.map_reduce(
    # map
    Code(" function() { \
        if(this.value != null) { \
            var newValue = parseInt(this.value.replace(\"$\", \"\").replace(\",\", \"\")); \
            emit(this.category, newValue); \
        } \
    }"),
    
    # reduce
    Code(" function(key, values) { \
        return Array.sum(values); \
    }"),
    
    # query output
    # Rezultat mapReduce będzie dostępny w kolekcji "results"
    "results"    
)
# Wyświetl zawartość "results"
for result in db.results.find({}):
    print(result)

File: Bazy-Danych/BazyDanych2/Lab4/zadanie domowe/zadanie3.js
db.question.mapReduce(
    //map
    function() {
        if(this.value != null) {
            // Usuń znak dolara i przecinka z pola "value", skonwertuj na inta,
            // a następnie prześlij do funkcji reduce
            var newValue = parseInt(this.value.replace("$", "").replace(",", ""));
            emit(this.category, newValue);
        }
    },
    
    //reduce
    function(key, values) {
        //Zsumuj wszystkie wartości
        return Array.sum(values);
    },
    
    //query output
    {
        //Rezultat mapReduce będzie dostępny w kolekcji "results"
        out: "results"
    }    
)
// Wyświetl zawartość "results"
db.results.find({})